ccp/src/prover.rs

```diff
@@ -67,10 +67,9 @@ impl NoxCCPApi for CCProver {
         new_epoch: EpochParameters,
         new_allocation: CUAllocation,
     ) -> Result<(), Self::Error> {
+        self.save_state(new_epoch, new_allocation.clone()).await?;
         self.apply_cc_parameters(new_epoch, &new_allocation).await?;
```

Comment:
I don't think that the current approach cause us any issues atm, but I think it'd be better to execute `apply_cc_parameters` first and only then save state, b/c it looks like an expected behavior and in future someone can rely on this and make some changes of core logic. So it could be smth like

```rust
        let result = self.apply_cc_parameters(new_epoch, &new_allocation).await;
        self.save_state(new_epoch, new_allocation.clone()).await?;
        result
```

Reply:
I think it can be further refined.  In your example, if both apply and save fail, save's error is returned.  I think apply error is more important.

However, this code is executed in background, so the error is only logged.  We might log both errors as well.

Reply:
ok, let's make the first error more important